sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-feedback-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,379 @@
+import {
+  ApplicationStatus,
+  COEStatus,
+  DisbursementScheduleStatus,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  createFileFromStructuredRecords,
+  getStructuredRecords,
+  mockDownloadFiles,
+} from "@sims/test-utils/mocks";
+import {
+  createTestingAppModule,
+  describeQueueProcessorRootTest,
+  mockBullJob,
+} from "../../../../../../test/helpers";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import { DeepMocked } from "@golevelup/ts-jest";
+import * as Client from "ssh2-sftp-client";
+import * as path from "path";
+import { FullTimeECertFeedbackIntegrationScheduler } from "../ecert-full-time-feedback-integration.scheduler";
+
+const FEEDBACK_ERROR_FILE_SINGLE_RECORD = "EDU.PBC.FTECERTSFB.SINGLERECORD";
+const FEEDBACK_ERROR_FILE_MULTIPLE_RECORDS =
+  "EDU.PBC.FTECERTSFB.MULTIPLERECORDS";
+const SHARED_DOCUMENT_NUMBER = 6666;
+const SHARED_ERROR_CODE = "EDU-00099";
+
+describe(
+  describeQueueProcessorRootTest(QueueNames.FullTimeFeedbackIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: FullTimeECertFeedbackIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+
+    beforeAll(async () => {
+      // Set the ESDC response folder to the mock folder.
+      process.env.ESDC_RESPONSE_FOLDER = path.join(
+        __dirname,
+        "e-cert-feedback-error-files",
+      );
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(FullTimeECertFeedbackIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Ensure the document number used along the tests will be unique.
+      await db.disbursementSchedule.update(
+        { documentNumber: SHARED_DOCUMENT_NUMBER },
+        { documentNumber: null },
+      );
+    });
+
+    it("Should log 'Invalid record type' error when the record type of header is invalid.", async () => {
+      // Arrange
+      mockDownloadFiles(
+        sftpClientMock,
+        [FEEDBACK_ERROR_FILE_SINGLE_RECORD],
+        (fileContent: string) => {
+          const file = getStructuredRecords(fileContent);
+          // Force the header to be wrong.
+          file.header = file.header.replace("100222  NEW", "500222  NEW");
```

Comment:
`replace` will change only the first match, hence the search string can be simplified as below.
```ts
file.header = file.header.replace("100", "500");
```
or
```ts
file.header = file.header.replace("1", "5");
```